xload/errors.go

```diff
@@ -61,3 +61,11 @@ type ErrInvalidPrefixAndKey struct {
 func (e ErrInvalidPrefixAndKey) Error() string {
 	return fmt.Sprintf("`%s` key=%s has both prefix and key", e.field, e.key)
 }
+
+// ErrCollision is returned when key collisions are detected.
+// Collision can happen when two or more fields have the same full key.
+type ErrCollision struct{ keys []string }
```

Comment:
Shall I return a copy of keys, or expose them directly?

| Direct | Copy |
|--------|--------|
| <pre>// ErrCollision is returned when key collisions are detected.<br>// Collision can happen when two or more fields have the same full key.<br>type ErrCollision struct{ Keys []string }</pre> | <pre>// ErrCollision is returned when key collisions are detected.<br>// Collision can happen when two or more fields have the same full key.<br>type ErrCollision struct{ keys []string }<br><br>// Keys returns the list of keys that caused the collision.<br>func (e *ErrCollision) Keys() []string {<br>&nbsp;&nbsp;&nbsp;&nbsp;keysCopy := make([]string, len(e.keys))<br>&nbsp;&nbsp;&nbsp;&nbsp;copy(keysCopy, e.keys)<br>&nbsp;&nbsp;&nbsp;&nbsp;return keysCopy<br>}</pre> |

Reply:
Direct is simpler and compatible with `errors.As`

Reply:
BTW `errors.As` works with private fields as well, making it compatible, will this change your opinion?

I'm ok with public slice as well, but all other errors are also following private variables. Since the ask was to allow users access to Keys slice, thought of just returning a copy for usage.

Reply:
Your call

Reply:
Keeping slice private for now to keep it in line with other errors, exposing a copied slice with a method for users.